hack/check-rec-rule-usage.sh

```diff
@@ -1,22 +1,41 @@
 #!/usr/bin/env bash
 set -e
-
 TMP=$(mktemp -d)
 echo "Created temporary directory at $TMP"
 
+CONOSLE_DIR=$1
+
+if [ ! -d $CONOSLE_DIR ]; then
+    echo "Directory $CONOSLE_DIR does not exist"
+    exit 1
+fi
+
 findrules="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)/find-rules.sh"
 
 "$findrules" >"$TMP/rule-groups"
 # extract recording rules
 tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.record) | .record' | tr -d '"' | sort -u >"$TMP/rec-rules"
+
 # and alerting rules
 tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.alert) | .expr' | grep "\w\+{" -o | tr -d "{" | sort -u >"$TMP/used-rules"
+
+# and referenced rules in other rules' expressions
+tmp/bin/gojsontoyaml -yamltojson <"$TMP/rule-groups" | jq -s '.. | objects | select(.expr) | .expr' | sed 's/\\n/\n/g' | sed 's/[^a-zA-Z0-9_]/\n/g' | sort -u | uniq >"$TMP/rule-expr-tokens"
+
+grep -Fxf "$TMP/rec-rules" "$TMP/rule-expr-tokens" || true >>"$TMP/used-rules"
+
 # and telemetry client's metrics
 tmp/bin/gojsontoyaml -yamltojson <manifests/*-config.yaml | jq '.data."metrics.yaml"' | sed "s;\\\\n;\n;g" | grep "^-\s'{__name__=\\\\\"" | sed -e "s/^- '{__name__=\\\\\"//" | sed -e "s/\\\\\"[^}]*}'//" | sort | uniq >>"$TMP/used-rules"
 
+tmp/bin/gojsontoyaml -yamltojson <manifests/*-config.yaml | jq '.data."metrics.yaml"' | sed "s;\\\\n;\n;g" | grep "^-\s'{__name__=~\\\\\"" | sed -e "s/^- '{__name__=~\\\\\"//" | sed -e "s/\\\\\"[^}]*}'//" | sort | uniq >"$TMP/used-rules-regex"
```

Comment:
can you add a comment to this line so that it is easier to understand what it does?